sklearn_numba_dpex/common/kernels.py

```diff
@@ -24,13 +24,20 @@
 import numba_dpex as dpex
 import numpy as np
 
-from sklearn_numba_dpex.common._utils import check_power_of_2
+from sklearn_numba_dpex.common._utils import (
+    _check_max_work_group_size,
+    check_power_of_2,
+)
 
 zero_idx = np.int64(0)
 
 
+# HACK: dtype argument is passed to prevent caching as a workaround for
```

Comment:
```suggestion
# HACK: dtype argument is passed to prevent sharing a device function
# instance between kernels specialized for different argument types.
# This is a workaround for:
```